### PR TITLE
Re-word UB in unsafe guide

### LINKED ISSUE #26346: TRPL is too soft on UB
from the unsafe section:

> Okay, let’s try again: what is not safe to do? [list of all UB in Rust]

Doing any of these things is *Undefined Behaviour* that should be impossible for a safe code to do *under any circumstance*. Code is *unsafe* if there are circumstances under which it may do one of these things (bad inputs, bad state).

### PR DESCRIPTION
**Pull request by steveklabnik**
This incorrectly implied that doing things is fine in unsafe code

Fixes #26346

### CODE REVIEW

File: src/doc/trpl/unsafe.md [hunk 1]
```diff
@@ -33,9 +33,21 @@ in the sections marked `unsafe`.
 
 # What does ‘safe’ mean?
 
-Safe, in the context of Rust, means “doesn’t do anything unsafe.” Easy!
+Safe, in the context of Rust, means ‘doesn’t do anything unsafe’. It’s also
+important to notice all kinds of behaviors that are certainly bad, but are
```
Comment (alexcrichton) [inline src/doc/trpl/unsafe.md]: s/all kinds/there are kinds/ ?

Comment (steveklabnik) [inline src/doc/trpl/unsafe.md]: updated, let me know what you think :)

**Comment by Gankro** [comment]
@bors r+

Thanks!

**Pull request merged**